### PR TITLE
Improve esbuild

### PR DESCRIPTION
**Pull request by lisez**
Support customized esbuild options from defineConfig. 

Such as `external` options to ignore esbuild to bundle some server-side modules.

And I found there is no format-rules in this repo, hence it changed a lot after I saved files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ If you wish to [use server-side rendering (SSR)](https://docs.astro.build/en/gui
 
 Add the Deno adapter to enable SSR in your Astro project with the following steps:
 
+1. Update or create `.npmrc` in your project with:
+
+   ```text
+   @jsr:registry=https://npm.jsr.io
+   ```
```
Comment (kt3k) [inline README.md:32]: This feels like an unnecessary friction to the users. Can we use other deep merge package which is available in npm?

**Review by kt3k — COMMENTED** [review]
Thanks for the PR! Mostly looks good to me. I left one question.

File: README.md (hunk 1 above)
Comment (lisez) [reply README.md]: @kt3k  I switched to lodash.

Comment (kt3k) [reply README.md]: Nice. Thanks!

**Review by kt3k — APPROVED** [review]
Thanks! LGTM

**Comment by kt3k** [comment]
published as `v0.1.3` https://www.npmjs.com/package/@deno/astro-adapter/v/0.1.3

**Pull request merged**